src/Modal.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import cx from 'classnames';
+import Icon from './Icon';
+import Button from './Button';
+
+/**
+ * SQ2 Modal component
+ * @see {@link http://meetup.github.io/sassquatch2/views.html#modals}
+ * @module Modal
+ */
+class Modal extends React.Component {
+	constructor(props){
+		super(props);
+		this.onDismiss = this.onDismiss.bind(this);
+		this.handleKeyDown = this.handleKeyDown.bind(this);
+	}
+
+	onDismiss(e) {
+		e.stopPropagation();
+
+		if (this.props.onDismiss) {
+			this.props.onDismiss(e);
+		}
+	}
+
+	handleKeyDown(e) {
+		if (e.key === 'Escape') {
+			this.onDismiss(e);
+		}
+	}
+
+	render() {
+		const {
+			className,
+			children,
+			fullscreen,
+			...other
+		} = this.props;
+
+		delete other.onDismiss; // onDismiss is consumed in this.onDismiss - do not pass it along to children
+
+		const classNames = cx(
+			className,
+			'modal'
+		);
+
+		const modalClasses = cx(
+			'view',
+			'padding--all',
+			{
+				'view--modalFull': fullscreen,
+				'view--modalSnap': !fullscreen
+			}
+		);
+
+		return (
+			<div
+				tabIndex='0'
+				onKeyDown={this.handleKeyDown}
```

Comment:
minor, but the convention for naming event handlers that are assigned to particular `onX` attributes in React is just to name them `onX` - `this.handleKeyDown` would be `this.onKeyDown`.